openedx/adg/lms/applications/test/test_api_views.py

```diff
@@ -0,0 +1,302 @@
+"""
+All tests for applications APIs views
+"""
+import json
+
+import pytest
+from django.urls import reverse
+from rest_framework import status
+
+from student.tests.factories import UserFactory
+
+from ..models import Education, WorkExperience
+from ..serializers import EducationSerializer, WorkExperienceSerializer
+from .factories import EducationFactory, UserApplicationFactory, WorkExperienceFactory
+
+JSON_CONTENT_TYPE = 'application/json'
+
+# pylint: disable=redefined-outer-name, unused-argument
+
+@pytest.fixture
+def user_instance(request):
+    """
+    Create user, this fixture will be passed as a parameter to all pytests
+    """
+    user = UserFactory()
+    user.set_password('password')
+    user.save()
+    return user
+
+
+@pytest.fixture
+def education_data(request, user_instance):
+    """
+    Create education data, this fixture will be passed as a parameter to all pytests
+    """
+    user_application = UserApplicationFactory(user=user_instance)
+    EducationFactory(user_application=user_application)
+    EducationFactory(user_application=user_application, degree=Education.DOCTORAL_DEGREE)
+
+
+@pytest.fixture
+def education_data_for_post_and_patch(request, education_data):
+    """
+    Create education data, this fixture will be passed as a parameter to all pytests
+    """
+    education = Education.objects.get(pk=1)
+    return {
+            'date_started_month': 2,
+            'date_started_year': 2018,
+            'date_completed_month': 8,
+            'date_completed_year': 2020,
+            'name_of_school': 'PUCIT',
+            'degree': Education.BACHELOR_DEGREE,
+            'area_of_study': 'Computer Sciences',
+            'is_in_progress': False,
+            'user_application': education.user_application.id
```

Comment:
We can, but in my opinion we shouldn't, we are returning a dictionary here that will be passed to our post and patch call. If we create an object and then create an dict out of it by initializing dict with object values, it will be overdoing a simple task.

Reply:
Ohk I got it. Thanks